extensions/standard-processors/processors/AttributesToJSON.cpp

```diff
@@ -0,0 +1,162 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "AttributesToJSON.h"
+
+#include "rapidjson/writer.h"
+#include "utils/StringUtils.h"
+#include "utils/ProcessorConfigUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+const std::set<std::string> AttributesToJSON::DESTINATIONS({"flowfile-attribute", "flowfile-content"});
+
+const core::Property AttributesToJSON::AttributesList(
+  core::PropertyBuilder::createProperty("Attributes List")
+    ->withDescription("Comma separated list of attributes to be included in the resulting JSON. "
+                      "If this value is left empty then all existing Attributes will be included. This list of attributes is case sensitive. "
+                      "If an attribute specified in the list is not found it will be be emitted to the resulting JSON with an empty string or NULL value.")
+    ->build());
+
+const core::Property AttributesToJSON::AttributesRegularExpression(
+  core::PropertyBuilder::createProperty("Attributes Regular Expression")
+    ->withDescription("Regular expression that will be evaluated against the flow file attributes to select the matching attributes. "
+                      "This property can be used in combination with the attributes list property.")
+    ->build());
+
+const core::Property AttributesToJSON::Destination(
+  core::PropertyBuilder::createProperty("Destination")
+    ->withDescription("Control if JSON value is written as a new flowfile attribute 'JSONAttributes' or written in the flowfile content. "
+                      "Writing to flowfile content will overwrite any existing flowfile content.")
+    ->isRequired(true)
+    ->withDefaultValue<std::string>("flowfile-attribute")
+    ->withAllowableValues<std::string>(DESTINATIONS)
+    ->build());
+
+const core::Property AttributesToJSON::IncludeCoreAttributes(
+  core::PropertyBuilder::createProperty("Include Core Attributes")
+    ->withDescription("Determines if the FlowFile core attributes which are contained in every FlowFile should be included in the final JSON value generated.")
+    ->isRequired(true)
+    ->withDefaultValue<bool>(true)
+    ->build());
+
+const core::Property AttributesToJSON::NullValue(
+  core::PropertyBuilder::createProperty("Null Value")
+    ->withDescription("If true a non existing or empty attribute will be NULL in the resulting JSON. If false an empty string will be placed in the JSON.")
+    ->isRequired(true)
+    ->withDefaultValue<bool>(false)
+    ->build());
+
+core::Relationship AttributesToJSON::Success("success", "All FlowFiles received are routed to success");
+
+void AttributesToJSON::initialize() {
+  setSupportedProperties({
+    AttributesList,
+    AttributesRegularExpression,
+    Destination,
+    IncludeCoreAttributes,
+    NullValue
+  });
+  setSupportedRelationships({Success});
```

Comment:
This processor has a failure relationship in NiFi.
Are you sure there is no such case where we can't provide proper json representation?

Reply:
I actually had a failure relationship, but it seemed to be forced and unnecessary in hindsight so I removed it in 74e8946b280c840c7c00e739f92bc5271ab69b11. NiFi has only one case where it transfers flowfile to failure and it's due to the used thirdparty JSON library's possible exception when writing the JSON object, but we do not have that here.

Reply:
Makes sense, thanks.